codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -314,7 +388,13 @@ private ChannelFuture finishEncode(final ChannelHandlerContext ctx, ChannelPromi
         final ByteBuf footer = ctx.alloc().heapBuffer(
                 compressor.maxCompressedLength(buffer.readableBytes()) + HEADER_LENGTH);
         flushBufferedData(footer);
+        if (bufferPromise != null) {
+            // buffer is empty now and its associated promise must be used with the write below
+            promise.addListener(new ChannelPromiseNotifier(bufferPromise));
+            bufferPromise = null;
+        }
 
+        footer.ensureWritable(HEADER_LENGTH);
```

Comment:
This looks like an unrelated bug. I had to add this because it kept randomly failing my new tests when compression would not save at least HEADER_LENGTH from the calculated max size above and so the header we write below wouldn't fit.

Reply:
@TommyWind could you open another PR with just this change ? This will make it easier for people to track bugs

Reply:
@normanmaurer sure, no problem :) here it is https://github.com/netty/netty/pull/11429